File: leetcode js/7. Reverse Integer.js
// 题目说明：
// Reverse digits of an integer.
// Example1: x = 123, return 321
// Example2: x = -123, return -321


/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    var string = x.toString(),
        array = string.split(""),
        len = array.length,
        i = len - 1,
        array2 = [],
        string2,
        num;

    for(;i >= 0 ;i--) {
        array2.push(array[i]);
    }

    string2 = array2.join("");
    num = parseInt(string2);
};

File: leetcode java/7. Reverse Integer.java
/*
思路：
首先，若要翻转，则循环就可以
num = num * 10 + x % 10;
x = x/10;
为了防止溢出，num定义为long
检查溢出，溢出返回0，没溢出返回num
 */
public class Solution {
    public int reverse(int x) {
        long num = 0;

        while(x != 0) {
            num = num * 10 + x % 10;
            x = x/10;
        }
        if(num >= Integer.MAX_VALUE || num <= Integer.MIN_VALUE) {
            return 0;
        }

        return (int)num;
    }
}
